subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -156,6 +168,44 @@ static int tls_entropy_func(void *ctx, unsigned char *buf, size_t len)
 }
 #endif /* defined(CONFIG_ENTROPY_HAS_DRIVER) */
 
+#if defined(CONFIG_NET_SOCKETS_ENABLE_DTLS)
+static void dtls_timing_set_delay(void *data, uint32_t int_ms, uint32_t fin_ms)
+{
+	struct dtls_timing_context *ctx = data;
+
+	ctx->int_ms = int_ms;
+	ctx->fin_ms = fin_ms;
+
+	if (fin_ms != 0) {
+		ctx->snapshot = k_uptime_get_32();
+	}
+}
+
+static int dtls_timing_get_delay(void *data)
+{
+	struct dtls_timing_context *timing = data;
+	unsigned long elapsed_ms;
+
+	NET_ASSERT(timing);
+
+	if (timing->fin_ms == 0) {
+		return -1;
+	}
+
+	elapsed_ms = k_uptime_get_32() - timing->snapshot;
+
+	if (elapsed_ms >= timing->fin_ms) {
+		return 2;
```

Comment:
Using constants for these return values would make it better to read

Reply:
These values come directly from mbedtls https://tls.mbed.org/api/ssl_8h.html#adea005600b532d788c05e9d01b42895b

I also used literal values in net_app in same function. There could be a comment here describing the values:
```
This callback must return:
       -1 if cancelled (fin_ms == 0),
        0 if none of the delays have passed,
        1 if only the intermediate delay has passed,
        2 if the final delay has passed.
```

Reply:
As this is an internal thing it's ok :)  Add this comment explaining these values is really nice !

Reply:
I'll add comment claryfing this.